AutomatedTesting/Gem/PythonTests/assetpipeline/scene_tests_fixtures/scene_test_debug_compare.py

```diff
@@ -0,0 +1,241 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+from __future__ import annotations
+
+import os
+import re
+from typing import List
+import ly_test_tools.o3de.pipeline_utils as utils
+import warnings
+import logging
+from automatedtesting_shared import asset_database_utils as asset_db_utils
+
+logger = logging.getLogger(__name__)
+
+
+def populate_expected_product_assets(assets_to_validate: List[asset_db_utils.DBSourceAsset]) -> List[str]:
+    """
+    Populate a list of expected product based on jobs completed for each source asset.
+    assets_to_validate: The expected job data for each source asset.
+    Example data:
+        "job_key": "Example Job",
+          "builder_guid": "Example_Guid",
+          "status": 4,
+          "error_count": 0,
+          "products": [
+            {
+              "product_name": "example_folder/example_asset.extension",
+              "sub_id": 0,
+              "asset_type": "Example_Asset_Type"
+            },
+    """
+    expected_product_list = []
+    for expected_source in assets_to_validate:
+        for expected_job in expected_source.jobs:
+            for expected_product in expected_job.products:
+                expected_product_list.append(expected_product.product_name)
+
+    return expected_product_list
+
+
+def trim_floating_point_values_from_same_length_lists(diff_actual: List[str], diff_expected: List[str],
+                                                      actual_file_path: str, expected_file_path: str) -> (List[str], List[str]):
+    """
+    This function will truncate any floating point values to be short enough to be stable.
+
+    o3de/o3de#13713: We need to stabilize the output of product assets, so this will emit a warning.
+
+    diff_actual: A list containing the debug output (dbgsg file) of the processed source asset under test.
+    diff_expected: A list containing the expected debug output of the source asset under test.
+    actual_file_path: Path to the dbgsg file containing the debug output of the processed source asset under test.
+    expected_file_path: Path to the dbgsg file containing the expected debug output to compare against.
+    """
+
+    # Get the initial list lengths, so they can be compared to the list lengths later to see if any differences were
+    # removed due to floating point value drift.
+    initial_diff_actual_len = len(diff_actual)
+    initial_diff_expected_len = len(diff_expected)
+
+    # Decide the number of digits past the decimal point to preserve.
+    DECIMAL_DIGITS_TO_PRESERVE = 3
```

Comment:
Nit: Could you add more comments to explain the reason for choosing 3 here?

Reply:
Yeah, I'll like this to a GHI as well.